### PR TITLE
fix(client): non-scalar sub selection for connections [BREAKING]

### PR DESCRIPTION
**Pull request by divyenduz**
* fix(client): non-scalar sub selection for connections - lib implementation
* fix(client): non-scalar sub selection for connections - types implementation

Fixes https://github.com/prisma/prisma/issues/3309

~~Note that - the existing types snapshot test cases do not cover this types change.~~ This is done now.

### CODE REVIEW

File: cli/packages/prisma-client-lib/src/Client.ts [hunk 1]
```diff
@@ -423,6 +424,14 @@ export class Client {
 
     const type = this.getDeepType(field.type)
 
+    if (isRelayConnection) {
+      let relayConnectionHasScalars = false
+      relayConnectionHasScalars = connectionNodeHasScalars({ type })
```
Comment (timsuchanek) [inline cli/packages/prisma-client-lib/src/Client.ts]: Please turn this into `const relayConnectionHasScalars = connectionNodeHasScalars({ type })`.

File: cli/packages/prisma-client-lib/src/utils/connectionNodeHasScalars.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        fieldType instanceof GraphQLScalarType ||
+        fieldType instanceof GraphQLEnumType
+    )
+}
+
+export function connectionNodeHasScalars({ type }) {
+    const edgesField = Object.entries(type.getFields())
+        .filter(([, subField]: any) => {
+            return subField.name === 'edges'
+        })
```
Comment (timsuchanek) [inline cli/packages/prisma-client-lib/src/utils/connectionNodeHasScalars.ts]: Please use `Object.values` here, then we can save the `.map` call https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values

File: cli/packages/prisma-client-lib/src/utils/connectionNodeHasScalars.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    if (edgesField.length === 0) {
+        return false
+    }
+
+    const edgesFieldType = getDeepType(edgesField[0].type)
+    const nodeField = Object.entries(edgesFieldType.getFields())
+        .filter(([, subField]: any) => {
+            return subField.name === 'node'
+        })
+        .map(([, subField]: any) => {
```
Comment (timsuchanek) [inline cli/packages/prisma-client-lib/src/utils/connectionNodeHasScalars.ts]: Here the same, please use `Object.values`

File: cli/packages/prisma-client-lib/src/utils/connectionNodeHasScalars.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            return subField
+        })
+    if (nodeField.length === 0) {
+        return false
+    }
+    const nodeFieldType = getDeepType(nodeField[0].type)
+    const nodeFieldScalars = Object.entries(nodeFieldType.getFields())
+        .filter(([, subField]: any) => {
+            return isScalar(subField)
+        })
```
Comment (timsuchanek) [inline cli/packages/prisma-client-lib/src/utils/connectionNodeHasScalars.ts]: `Object.values`

**Comment by divyenduz** [comment]
Thanks. I made the required changes :)

**Pull request merged**